requirements-dev.txt

```diff
@@ -1,5 +1,5 @@
 mypy==0.812
-pylint==2.4.4
+pylint==2.12.1
```

Comment:
`pylint 2.4.4` was released about 2 years ago. I believe some of the best practices have changed since that time as the APIs have evolved.

Used the update to Python 3.9 as a trigger to switch to the most recent version. Was forced to do so because `pylint 2.4.4` reported many false  positives with `Python 3.9` (see https://github.com/PyCQA/pylint/issues/3882 for details).